docs/user/troubleshooting_job.md

```diff
@@ -17,51 +17,197 @@
 
-         ```sh
-         chmod 400 userName~jobName
-         ```
+### Validate with local simulator
 
-      5. Use `ssh` command to connect into container. for example
+Some bugs may happen in specific environment or configuration, so the code may run well at local, but failed remotely. If the environment can be simulated locally, more environment issues can be found.
 
-         ```sh
-         ssh -i userName~jobName -p ssh_port root@container_ip
-         ```
+OpenPAI VS Code Client can consume OpenPAI job configuration file and run in Docker container locally. This simulation can find problems that related to job configuration files, mismatched docker image with code dependencies, command line errors. It can cover most situations at remote, but still limited, like the requested resource in configuration is ignored, as in most case, local machine is not powerful like a GPU server. So, when code is running locally, it may be much slower, and may be out of memory. The code or command can be modified to avoid this kind of issues and reduce training times to disclose more remote problems.
 
-    - SSH connect inside containers
+The simulator in OpenPAI VS Code Client needs Docker installed. Refer to [here](../../contrib/pai_vscode/VSCodeExt.md) to install and learn how to [simulate Job Running](../../contrib/pai_vscode/README.md#simulate-job-running).
 
-      You can use `ssh $PAI_CURRENT_TASK_ROLE_NAME-$PAI_CURRENT_TASK_ROLE_CURRENT_TASK_INDEX` command to connect into another containers which belong to the same job. For example, if there are two taskRoles: master and worker, you can connect to worker-0 container directly with below command line:
+Note, as Docker doesn't support GPU on Windows, so it may need a docker image with CPU package for local simulation.
 
-      ```sh
-      ssh worker-0
-      ```
+### Know resource bottleneck well
 
-1. Job Profiling
+To use OpenPAI, user needs to specify used resources, including CPU, GPU and memory. If requested resource is lower than minimum requirement, the progress may be much slower than expected. So, to be aware and understand bottleneck is important for efficiency.
 
-      Users can view the resource cost and bottlenecks of various metrics of the job by following the [job profiling doc](../job_profiling.md).
+OpenPAI provides metrics of CPU, memory, and GPU. Refer to [how to check job metrics](#how-to-check-job-metrics) in this document to learn more.
+
+## Diagnostic issues
+
+### Job is waiting for hours
+
+In general, jobs of OpenPAI stays in waiting status about 1 minute. But if there is not enough resource, a job may stay in waiting status longer.
+
+![waiting](imgs/web_job_list_waiting.png)
+
+In this case, the job is in queue to wait resource. When some jobs complete, this job will get resource to run. Another way is to reduce requested resource, so this job may get enough resource easier.
```

Comment:
Can direct user to virtual cluster page to let user checks if requesting resource can not be satisfied by this VC.

Reply:
Also #2208 will be useful for user to debug.

Reply:
VC is an advanced topic for admin. A cluster may enable or disable preemption, it's hard to let user to judge it, and even in the virtual cluster, fragments resource still blocks allocation. The goal is to let user know fragmentation may block resource allocation, that's enough to a user.

Reply:
Ok, maybe we can change the doc later when #2208 has finished.